Mark the ts auto type generation as experimental (#1767)
strapi/documentation: docusaurus/docs/dev-docs/configurations/typescript.md

| ----------------------------------------- | ---------------- | ------------------------------------------------------------------------------------------------------------------------------------------------ |
| `./dist` directory                        | application root | Adds the location for compiling the project JavaScript source code.                                                                              |
| `build` directory                         | `./dist`         | Contains the compiled administration panel JavaScript source code. The directory is created on the first `yarn build` or `npm run build` command |
| `tsconfig.json` file                      | application root | Manages TypeScript compilation for the server.                                                                                                   |
| `tsconfig.json` file                      | `./src/admin/`   | Manages TypeScript compilation for the admin panel.                                                                                              |

## Strapi-specific configuration for TypeScript

:::caution 🚧 This feature is considered experimental.

> These settings should be reserved to an experimental usage, as they might have issues or break some features.

Reviewer: ```suggestion
These settings are considered experimental and might have issues or break some features.
```
